pennylane/_qubit_device.py

```diff
@@ -1000,6 +1031,8 @@ def sample(self, observable, shot_range=None, bin_size=None):
                 ) from e
 
         if bin_size is None:
+            if counts:
```

Comment:
What happens if `bin_size is not None` and `counts==True`? Is that case supported?

Reply:
I have doubts about that case. We can count within bins and return a list of dicts, but I don't quite see why it may be useful. Another option is to ignore `bin_size` if `counts=True` is provided and return one dict.
What is the best option?

Reply:
As a test version I implemented grouping bins by counts